core/src/main/java/org/infinispan/interceptors/distribution/BaseDistributionInterceptor.java

```diff
@@ -253,16 +253,7 @@ private boolean shouldLoad(InvocationContext ctx, AbstractDataWriteCommand comma
             case DONT_LOAD:
                return false;
             case OWNER:
-               switch (info.ownership()) {
-                  case PRIMARY:
-                     return true;
-                  case BACKUP:
-                     return !ctx.isOriginLocal();
-                  case NON_OWNER:
-                     return false;
-                  default:
-                     throw new IllegalStateException();
-               }
+               return info.isPrimary() || info.isWriteOwner() && !ctx.isOriginLocal();
```

Comment:
I don't know about you, but for me mapping `||` to `+` and `&&` to `*` is so ingrained that I find parentheses annoying in expressions like this.